Assets/Scripts/RoadSegment.cs

```diff
@@ -0,0 +1,14 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RoadSegment
+{
+    private float edgeNearZ, edgeFarZ, curve, y;
+    public float EdgeNearZ { get => edgeNearZ; set => edgeNearZ = value; }
+    public float EdgeFarZ { get => edgeFarZ; set => edgeFarZ = value; }
+    public float Curve { get => curve; set => curve = value; }
+    public float Y { get => y; set => y = value; }
```

Comment:
I notice you are using public properties along with public getters and setters.

Perhaps you meant to make the setters private like this:
```
public float edgeNearZ { get; private set; }
public float edgeFarZ {get; private set; }
public float curve { get; private set; }
public float y {get; private set; }
```

This would keep these members as read-only to every other object, leaving write privileges to only this instance of RoadSegment.

Alternatively, if you realize that you do need other objects to be able to write to these values as well, you can just use this for everything:
`public float edgeNearZ, edgeFarZ, curve, y;`

There are also ways to make write privileges available to other select objects, but that is a bit more involved ('Friend' objects don't exist naturally in C#).